pandas/tests/test_resample.py

```diff
@@ -2799,89 +2693,182 @@ def test_evenly_divisible_with_no_extra_bins(self):
         result = df.resample('7D').sum()
         assert_frame_equal(result, expected)
 
-    def test_loffset_returns_datetimeindex(self):
+
+def assert_series_or_frame_equal(result, expected):
+    if isinstance(result, Series):
+        return assert_series_equal(result, expected)
+    else:
+        return assert_frame_equal(result, expected)
+
+
+class TestPeriodIndexIdiomaticPytest:
+
+    @pytest.fixture(scope='class')
+    def index(self):
+        return period_range(datetime(2005, 1, 1),
+                            datetime(2005, 1, 10), freq='D')
+
+    @pytest.fixture(scope='class')
+    def series(self, index):
+        return Series(np.arange(len(index)), index=index, name='pi')
+
+    @pytest.fixture(scope='class')
+    def frame(self, index):
+        return DataFrame({'value': np.arange(len(index))}, index=index)
+
+    @pytest.fixture(params=[Series, DataFrame], scope='class')
+    def pandas_obj(self, request, index):
```

Comment:
sorry u meant the Base class and not TestCase base class

in any event it's not more clear to me